libr/parse/code.c

```diff
@@ -71,7 +71,7 @@ static int __typeLoad(void *p, const char *k, const char *v) {
 		}
 		tcc_sym_push ((char *)typename, typesize, btype);
 	}
-	return 0;
+	return true;
```

Comment:
this condition seems inverted. so the behaviour has changed

Reply:
if everything went well, it should return true, not false. before false was `-1` and true was `0` in this func

Reply:
yeah i noticed that while writing, however the logic in this function seems a bit weird, as it should `return false` so it would just get the first element.

Reply:
Indeed I think this code is never really executed...  See https://codecov.io/gh/radareorg/radare2/src/fab89a61d9bc969c15d006e950eb0e4f2d5101bd/libr/parse/code.c#L26 . The real part of the function is never tested and I doubt it worked well, because the loop was interrupted immediately due to this `return 0`.

We have 2 options:
1) adjust the tests to include the `Not found` message
2) remove the `Not found` message from tccgen.c

Preferences?

Reply:
2

Reply:
2